### ISSUE TITLE
BUG: the precision of big integer in read_json

### ISSUE THREAD

**Issue by bytedynamic** [pandas-dev/pandas#44684]
### 

- [X] I have checked that this issue has not already been reported.

- [X] I have confirmed this bug exists on the [latest version](https://pandas.pydata.org/docs/whatsnew/index.html) of pandas.

- [ ] I have confirmed this bug exists on the master branch of pandas.

### Reproducible Example

```python
import pandas as pd

jsn = '[{"NO": "202101010000000003", "NAME": "A"}, {"NO": "202101010000000013", "NAME": "B"}, {"NO": "202101010000000023", "NAME": "C"}]'


df1 = pd.read_json(jsn)
print(df1)
```

### Issue Description

I have a json file, and column "NO" is a big integer.
I use pd.read_json() read the json file, however, the  value of column "NO" in the DataFrame is not exactly equal to the value in json file. DataFrame as follows:

| | NO | NAME
-- | -- | --
0 | 202101010000000000 | A
1 | 202101010000000000 | B
2 | 202101010000000032 | C

and I use df.info(), which shows the dtype of Column "NO" is int64.
```
<class 'pandas.core.frame.DataFrame'>
RangeIndex: 3 entries, 0 to 2
Data columns (total 2 columns):
 #   Column  Non-Null Count  Dtype 
---  ------  --------------  ----- 
 0   NO      3 non-null      int64 
 1   NAME    3 non-null      object
dtypes: int64(1), object(1)
memory usage: 176.0+ bytes
```

It's really strange. The range of int64 is -9223372036854775808 ~ 9223372036854775807 and 202101010000000000 is in the range.

### Expected Behavior

| | NO | NAME
-- | -- | --
0 | 202101010000000003 | A
1 | 202101010000000013 | B
2 | 202101010000000023 | C

### Installed Versions

<details>

INSTALLED VERSIONS
------------------
commit           : 945c9ed766a61c7d2c0a7cbb251b6edebf9cb7d5
python           : 3.8.8.final.0
python-bits      : 64
OS               : Windows
OS-release       : 10
Version          : 10.0.19041
machine          : AMD64
processor        : Intel64 Family 6 Model 94 Stepping 3, GenuineIntel
byteorder        : little
LC_ALL           : None
LANG             : None
LOCALE           : Chinese (Simplified)_China.936

pandas           : 1.3.4
numpy            : 1.21.4
pytz             : 2021.1
dateutil         : 2.8.1
pip              : 21.3.1
setuptools       : 52.0.0.post20210125
Cython           : 0.29.23
pytest           : 6.2.3
hypothesis       : None
sphinx           : 4.0.1
blosc            : None
feather          : None
xlsxwriter       : 1.3.8
lxml.etree       : 4.6.3
html5lib         : 1.1
pymysql          : 1.0.2
psycopg2         : None
jinja2           : 2.11.3
IPython          : 7.22.0
pandas_datareader: None
bs4              : 4.9.3
bottleneck       : 1.3.2
fsspec           : 0.9.0
fastparquet      : None
gcsfs            : None
matplotlib       : 3.3.4
numexpr          : 2.7.3
odfpy            : None
openpyxl         : 3.0.7
pandas_gbq       : None
pyarrow          : 5.0.0
pyxlsb           : None
s3fs             : None
scipy            : 1.6.2
sqlalchemy       : 1.4.7
tables           : 3.6.1
tabulate         : None
xarray           : 0.20.1
xlrd             : 2.0.1
xlwt             : 1.3.0
numba            : 0.53.1
</details>

**Comment by AlexKirko** [comment]
Hello, @bytedynamic ! Thanks for reporting this.

Looks like it's a data type inferrence issue, because this works perfectly:

```python
import pandas as pd

jsn = '[{"NO": "202101010000000003", "NAME": "A"}, {"NO": "202101010000000013", "NAME": "B"}, {"NO": "202101010000000023", "NAME": "C"}]'


df1 = pd.read_json(jsn, dtype={"NO": pd.Int64Dtype(), "NAME": str})
print(df1)
```

| |NO|NAME|
|-|-|-|
|0|202101010000000003|A|
|1|202101010000000013|B|
|2|202101010000000023|C|

**Comment by bytedynamic** [comment]
> Hello, @bytedynamic ! Thanks for reporting this.
> 
> Looks like it's a data type inferrence issue, because this works perfectly:
> 
> ```python
> import pandas as pd
> 
> jsn = '[{"NO": "202101010000000003", "NAME": "A"}, {"NO": "202101010000000013", "NAME": "B"}, {"NO": "202101010000000023", "NAME": "C"}]'
> 
> 
> df1 = pd.read_json(jsn, dtype={"NO": pd.Int64Dtype(), "NAME": str})
> print(df1)
> ```
> 
> NO	NAME
> 0	202101010000000003	A
> 1	202101010000000013	B
> 2	202101010000000023	C

Yeah, if I use `read_json(jsn, dtype=False)` to read json file, and the dataframe is same as your result. So I think it might be a bug in data type transfer in `read_json`.
